src/NodaTime.Benchmarks.Custom/ConcurrentNewString.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright 2023 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using System.Threading;
+
+namespace NodaTime.Benchmarks.Custom;
+
+internal class ConcurrentNewString : ConcurrencyCommandBase
+{
+    private ConcurrentNewString(int threadCount, Duration executionTime) : base(threadCount, executionTime)
+    {
+    }
+
+    protected override void DoWork()
+    {
+        while (Interlocked.Add(ref iterations, ChunkSize) > 0)
+        {
+            for (int i = 0; i < ChunkSize; i++)
+            {
+                _ = new string('x', 8);
```

Comment:
Also, I'm somewhat surprised these benchmarks don't end up getting optimized out of existence (either at the IL level or when JITted).